tests/src/test/java/alluxio/server/ft/journal/raft/EmbeddedJournalIntegrationTestFaultTolerance.java

```diff
@@ -0,0 +1,305 @@
+  }
+
+  private void waitForSnapshot(File raftDir) throws InterruptedException, TimeoutException {
+    File snapshotDir = new File(raftDir, "sm");
+    final int RETRY_INTERVAL_MS = 200; // milliseconds
+    CommonUtils.waitFor("snapshot is downloaded", () -> {
+      File[] files = snapshotDir.listFiles();
+      return files != null && files.length > 1 && files[0].length() > 0;
+    }, WaitForOptions.defaults().setInterval(RETRY_INTERVAL_MS).setTimeoutMs(RESTART_TIMEOUT_MS));
+  }
+
+  @Test
+  public void restart() throws Exception {
+    mCluster = MultiProcessCluster.newBuilder(PortCoordination.EMBEDDED_JOURNAL_RESTART)
+        .setClusterName("EmbeddedJournalFaultTolerance_restart")
+        .setNumMasters(NUM_MASTERS)
+        .setNumWorkers(NUM_WORKERS)
+        .addProperty(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.EMBEDDED.toString())
+        .addProperty(PropertyKey.MASTER_JOURNAL_FLUSH_TIMEOUT_MS, "5min")
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT, "750ms")
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT, "1500ms")
+        .build();
+    mCluster.start();
+
+    AlluxioURI testDir = new AlluxioURI("/dir");
+    FileSystem fs = mCluster.getFileSystemClient();
+    fs.createDirectory(testDir);
+    restartMasters();
+    assertTrue(fs.exists(testDir));
+    restartMasters();
+    assertTrue(fs.exists(testDir));
+    restartMasters();
+    assertTrue(fs.exists(testDir));
+    mCluster.notifySuccess();
+  }
+
+  @Test
+  public void restartStress() throws Throwable {
+    mCluster = MultiProcessCluster.newBuilder(PortCoordination.EMBEDDED_JOURNAL_RESTART_STRESS)
+        .setClusterName("EmbeddedJournalFaultTolerance_restartStress")
+        .setNumMasters(NUM_MASTERS)
+        .setNumWorkers(NUM_WORKERS)
+        .addProperty(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.EMBEDDED.toString())
+        .addProperty(PropertyKey.MASTER_JOURNAL_FLUSH_TIMEOUT_MS, "5min")
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT, "750ms")
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT, "1500ms")
+        .build();
+    mCluster.start();
+
+    // Run and verify operations while restarting the cluster multiple times.
+    AtomicReference<Throwable> failure = new AtomicReference<>();
+    AtomicInteger successes = new AtomicInteger(0);
+    FileSystem fs = mCluster.getFileSystemClient();
+    List<EmbeddedJournalIntegrationTestFaultTolerance.OperationThread> threads = new ArrayList<>();
+    try {
+      for (int i = 0; i < 10; i++) {
+        EmbeddedJournalIntegrationTestFaultTolerance.OperationThread t =
+            new EmbeddedJournalIntegrationTestFaultTolerance
+                .OperationThread(fs, i, failure, successes);
+        t.start();
+        threads.add(t);
+      }
+      for (int i = 0; i < 2; i++) {
+        restartMasters();
+        System.out.printf("---------- Iteration %s ----------\n", i);
```

Comment:
remove debugging printf?

Reply:
fixed, this was a direct port from the previous `EmbeddedJournalIntegrationTest` file